examples/download_sample_data.py

```diff
@@ -217,6 +233,40 @@ def _process_dataset(dataset, output_dir):
                 zipf.extractall()
             os.remove(output_path)
 
+
+def _unzip_7z(fname, delete_7z=True):
+    try:
+        arc = Archive7z(open(fname, 'rb'))
+    except:
+        print('FAILED ON 7Z', fname)
+        raise
+    fnames = arc.filenames
+    files = arc.files
+    data_dir = os.path.dirname(fname)
+    for fn, fi in zip(fnames, files):
+        gnd = path.join(data_dir, os.path.basename(fn))
+        if not os.path.exists(os.path.dirname(gnd)):
+            os.mkdir(os.path.dirname(gnd))
+        with open(gnd, 'w') as f:
+            f.write(fi.read().decode())
+        if delete_7z:
+            os.remove(fname)
+
+
+def download_puget_sound_lidar(here, files, url, **dataset):
+    urls = [url.replace('.html', '/' + _.strip())
+            for _ in files]
+    output_paths = [path.join(here, 'data', os.path.basename(fname))
+                    for fname in files]
+    title_fmt = 'Puget Sound Lidar {} of {}'
+    for idx, (url, output_path) in enumerate(zip(urls, output_paths)):
+        if os.path.exists(output_path.replace('7z', 'gnd')):
+            print('Skipping {0}'.format(title_fmt.format(idx + 1, len(urls))))
+            continue
+        _url_to_binary_write(url, output_path, title_fmt.format(idx + 1, len(urls)))
+        _unzip_7z(output_path, delete_7z=True)
+
```

Comment:
It's a shame that we need such dataset-specific code in here.  I haven't studied it to see how it works and precisely what it is accomplishing, but maybe it could be converted into a general function that accepts an argument, with the argument supplied based on data in the datasets.yml file?  I.e., perhaps change "title" in the .yml from "Puget Sound LiDAR Consortium" to "Puget Sound LiDAR", and then rename this function (and add a docstring) to explain concretely what it's doing with the files, rather than just saying it's about lidar and leaving it mysterious.